### PR TITLE
Fix feature gate only being checked on first repr attr.

### PR DESCRIPTION
**Pull request by bitshifter**
Reported in https://github.com/rust-lang/rust/issues/33158#issuecomment-412185357.

### CODE REVIEW

File: src/libsyntax/feature_gate.rs [hunk 1]
```diff
@@ -1625,7 +1625,7 @@ impl<'a> Visitor<'a> for PostExpansionVisitor<'a> {
             }
 
             ast::ItemKind::Struct(..) => {
-                if let Some(attr) = attr::find_by_name(&i.attrs[..], "repr") {
+                for attr in attr::filter_by_name(&i.attrs[..], "repr") {
```
Comment (RalfJung) [inline src/libsyntax/feature_gate.rs]: This makes me wonder in how many other cases we have the same problem...

**Comment by kennytm** [comment]
@bors retry 

30 min timeout

File: src/libsyntax/feature_gate.rs (hunk 1 above)
Comment (bitshifter) [reply src/libsyntax/feature_gate.rs]: This was the only one that looked like a problem for `repr`. I don't know about other attributes though.

**Pull request merged**